stwcs/updatewcs/astrometry_utils.py

```diff
@@ -580,44 +581,116 @@ def apply_new_apriori(self, obsname):
 
         return wname
 
-    def isAvailable(self):
-        """Test availability of astrometryDB web-service."""
+    def isAvailable(self, max_tries=2, testing=False):
+        """Tests the availability of the astrometry database
+
+        Parameters
+        ----------
+        max_tries : int, optional
+            Number of attempts to query the db, by default 2
+        testing : bool, optional
+            If True, use a very short timeout for testing purposes, by default False
+
+        Raises
+        ------
+        ConnectionRefusedError
+            If the service is unavailable after max_tries attempts and raise_errors is True
+        ConnectionError
+            If there is a network-related error during the request and raise_errors is True
+        """
         if not self.perform_step:
             return
 
-        serviceEndPoint = self.serviceLocation + 'availability'
-        logger.info(f"AstrometryDB URL: {serviceEndPoint}")
+        # Set timeout based on testing flag
+        timeout = 1e-15 if testing else 5.0
```

Comment:
Presumably in seconds. Can you add a comment?

Reply:
Added with commit [ed5b1e2b5993bdeeacb3bace632da26ec7fccf8c](https://github.com/spacetelescope/stwcs/pull/231/commits/ed5b1e2b5993bdeeacb3bace632da26ec7fccf8c)